File: src/arabagent/Node.java
package arabagent;

import java.util.*;

public class Node {

    String key;
    double weight;

    int totalNumOfDocs, totalNumOfDays;

    HashMap<String, Double> neigbours;

    public Node(String con) { key = con; neigbours = new HashMap<String, Double>(); }
    public Node(String con, double imp) { key = con; weight = imp; neigbours = new HashMap<String, Double>(); }
    public Node(String con, double imp, int docNo) { key = con; weight = imp; neigbours = new HashMap<String, Double>(); totalNumOfDocs = docNo; }

    public void addNewNeigbour(String nKey)
    {
        neigbours.put(nKey, 0.0);
    }

    public void addNewNeigbour(String nKey, double co_ocurr)
    {
        neigbours.put(nKey, co_ocurr);
    }

    public boolean isNeighbour(String nKey)
    {
        return neigbours.containsKey(nKey);
    }

    public double relationWeight(String nKey)
    {
        return neigbours.get(nKey);
    }

    //properties
    public String getNodeKey(){
        return key;
    }
    
    public void setNodeWeight(double weight){
        this.weight = weight;
    }
    
    public double getNodeWeight(){
        return weight;
    }

    public void setDocFreq(int totalNumOfDocs){
        this.totalNumOfDocs = totalNumOfDocs; 
    }
    
    public int getDocFreq(){
        return totalNumOfDocs;
    }
}

File: dosDataStructure/Node.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace dosDataStructure
{
    public class Node
    {
        string key;
        double weight;

        int totalNumOfDocs, totalNumOfDays;

        Dictionary<string, double> neigbours;

        public Node(string con) { key = con; neigbours = new Dictionary<string, double>(); }
        public Node(string con, double imp) { key = con; weight = imp; neigbours = new Dictionary<string, double>(); }
        public Node(string con, double imp, int docNo) { key = con; weight = imp; neigbours = new Dictionary<string, double>(); totalNumOfDocs = docNo; }

        public void AddNewNeigbour(string nKey)
        {
            neigbours.Add(nKey, 0);
        }

        public void AddNewNeigbour(string nKey, double co_ocurr)
        {
            neigbours.Add(nKey, co_ocurr);
        }

        public bool isNeighbour(string nKey)
        {
            return neigbours.ContainsKey(nKey);
        }

        public double relationWeight(string nKey)
        {
            return neigbours[nKey];
        }

        //properties
        public string NodeKey
        {
            get { return key; }
        }
        public double NodeWeight
        {
            get { return weight; }
            set { weight = value; }
        }

        public int DocFreq
        {
            get { return totalNumOfDocs; }
            set { totalNumOfDocs = value; }
        }

        //public int DayFreq
        //{
        //    get { return totalNumOfDays; }
        //    set { totalNumOfDays = value; }
        //}
    }
}
